src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -455,7 +455,7 @@ public ReadOnlySequence<T> Slice(SequencePosition start, SequencePosition end)
         public ReadOnlySequence<T> Slice(SequencePosition start)
         {
             BoundsCheck(start);
-            return SliceImpl(start);
+            return SliceImpl(start.Equals(default) ? Start : start);
```

Comment:
As far as I recall, this code is quite perf-sensitive. Are we OK with introducing branches with checks like these?

Maybe `start.GetObject() == null` instead?

It's probably not worth it, but consider even calling a helper that takes object/int (and passing in `_startObject`, `_startInteger` rather than creating a SequencePosition to pass in which gets deconstructed anyway).

cc @davidfowl, @pakrym